src/main/java/com/recap/updater/holdings/KinesisProcessor.java

```diff
@@ -0,0 +1,95 @@
+package com.recap.updater.holdings;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.kinesis.model.PutRecordsRequest;
+import com.amazonaws.services.kinesis.model.PutRecordsRequestEntry;
+import com.amazonaws.services.kinesis.model.PutRecordsResult;
+import com.amazonaws.services.kinesis.model.PutRecordsResultEntry;
+import com.recap.config.BaseConfig;
+import com.recap.config.EnvironmentConfig;
+import com.recap.constants.Constants;
+import com.recap.exceptions.RecapHarvesterException;
+
+public class KinesisProcessor implements Processor {
+
+  private BaseConfig baseConfig;
+
+  private static Logger logger = LoggerFactory.getLogger(KinesisProcessor.class);
+
+  public KinesisProcessor(BaseConfig baseConfig) {
+    this.baseConfig = baseConfig;
+  }
+
+  @Override
+  public void process(Exchange exchange) throws RecapHarvesterException {
+    try {
+      List<byte[]> avroItems = exchange.getIn().getBody(List.class);
+      if (avroItems.size() > Constants.KINESIS_PUT_RECORDS_MAX_SIZE) {
+        splitItemsAndSendToKinesis(avroItems);
+      } else
+        sendToKinesis(avroItems);
+    } catch (Exception e) {
+      logger.error("Error occurred while sending items to kinesis - ", e);
+      throw new RecapHarvesterException(
+          "Error occurred while sending records to kinesis - " + e.getMessage());
+    }
+  }
+
```

Comment:
Great suggestion. I am going to take a look at this utility and see if we can use this here.